sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala

```diff
@@ -36,6 +37,7 @@ class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
    * @since 2.0.0
    */
   def set(key: String, value: String): Unit = {
+    assertNotGlobalSQLConf(key)
```

Comment:
I am wondering if we need to add this `assertNotGlobalSQLConf ` in all the public set functions, right? I am just afraid we might forget to update the other `set` functions when adding more global sqlconf.